components/base/wheeled/kinematics.go

```diff
@@ -0,0 +1,117 @@
+	"go.viam.com/rdk/spatialmath"
+)
+
+type kinematicWheeledBase struct {
+	*wheeledBase
+	slam  slam.Service
+	model referenceframe.Model
+}
+
+// WrapWithKinematics takes a wheeledBase component and adds a slam service to it
+// It also adds kinematic model so that it can be controlled.
+func WrapWithKinematics(ctx context.Context, base *wheeledBase, slam slam.Service) (base.KinematicBase, error) {
+	var err error
+	kwb := &kinematicWheeledBase{
+		wheeledBase: base,
+		slam:        slam,
+	}
+	kwb.model, err = kwb.buildModel(ctx)
+	if err != nil {
+		return nil, err
+	}
+	return kwb, err
+}
+
+func (kwb *kinematicWheeledBase) ModelFrame() referenceframe.Model {
+	return kwb.model
+}
+
+func (kwb *kinematicWheeledBase) CurrentInputs(ctx context.Context) ([]referenceframe.Input, error) {
+	// TODO(RSDK-2311): complete the implementation
+	return []referenceframe.Input{}, errors.New("not implemented yet")
+}
+
+func (kwb *kinematicWheeledBase) GoToInputs(ctx context.Context, goal []referenceframe.Input) error {
+	// TODO(RSDK-2311): complete the implementation
+	return errors.New("not implemented yet")
+}
+
+// ModelFrame builds the kinematic model associated with the kinematicWheeledBase
+// Note that this model is not intended to be registered in the frame system
+func (kwb *kinematicWheeledBase) buildModel(ctx context.Context) (referenceframe.Model, error) {
+	if kwb.collisionGeometry == nil {
+		return nil, errors.New("cannot create model for base with no collision geometry")
+	}
+
+	// get the limits of the SLAM map to set as the extents of the frame
+	data, err := slam.GetPointCloudMapFull(ctx, kwb.slam, "")
+	if err != nil {
+		return nil, err
+	}
+	pcd, err := pointcloud.ReadPCD(bytes.NewReader(data))
+	if err != nil {
+		return nil, err
+	}
+	dims := pcd.MetaData()
+
+	// build the model - SLAM convention is that the XZ plane is the ground plane
+	frame2D, err := referenceframe.NewMobile2DFrame(
+		kwb.collisionGeometry.Label(),
+		[]referenceframe.Limit{{Min: dims.MinX, Max: dims.MaxX}, {Min: dims.MinZ, Max: dims.MaxZ}},
+		kwb.collisionGeometry)
+	if err != nil {
+		return nil, err
+	}
+	model := referenceframe.NewSimpleModel(kwb.name)
+	model.OrdTransforms = []referenceframe.Frame{frame2D}
+	return model, nil
+}
+
+func collisionGeometry(cfg config.Component) (spatialmath.Geometry, error) {
+	// TODO(rb): this is a hacky workaround for not having kinematics for bases yet
+	// we create a sphere that would encompass the config geometry's rotation a full 360 degrees
+	// TODO(RSDK-1014): the orientation of this model will matter for collision checking,
+	// and should match the convention of +Y being forward for bases
+	geoCfg := cfg.Frame.Geometry
+	r := geoCfg.TranslationOffset.Norm()
+	switch geoCfg.Type {
+	case spatialmath.BoxType:
+		r += r3.Vector{X: geoCfg.X, Y: geoCfg.Y, Z: geoCfg.Z}.Norm() / 2
+	case spatialmath.SphereType:
+		r += geoCfg.R
+	case spatialmath.CapsuleType:
+		r += geoCfg.L / 2
+	case spatialmath.UnknownType:
```

Comment:
Why isn't this an error? Why would the geometry have unknown type, but include parameters from one of the supported types?

Reply:
We don't require users to specify the geometry type in the config.  If they do not, we do our best to parse it and infer which they meant.  This snippet is more or less copy/pasted from `ParseConfig` in `geometry.go`.  

I call out in a comment at the top of the function that this is hacky; copy pasting from spatialmath is not something we want to do.  But this will only exist until we figure out how we want to support kinematics.  At which time it will go away in favor of just using the user supplied geometry as collision information

Reply:
thank you!